rust/core-lib/src/client.rs

```diff
@@ -36,6 +36,18 @@ pub struct WidthReq {
     pub strings: Vec<String>,
 }
 
+#[derive(Serialize, Deserialize)]
+pub struct Range {
```

Comment:
What is this used for, again? As in, if this is present what is the client supposed to do with that information?

Reply:
The client is supposed to highlight this area while hover is active. Thinking again, it makes sense to convert this to Line/Col according to View which client can use to have a tint over that area. What do you think?

Reply:
@cmyr Views on this?